tests/models/rt_detr/test_modeling_rt_detr.py

```diff
@@ -0,0 +1,457 @@
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Testing suite for the PyTorch RT_DETR model. """
+
+
+import inspect
+import unittest
+
+from transformers import RTDetrConfig, RTDetrImageProcessor, is_torch_available, is_vision_available
+from transformers.testing_utils import require_torch, require_vision, torch_device
+from transformers.utils import cached_property
+
+from ...test_configuration_common import ConfigTester
+from ...test_modeling_common import ModelTesterMixin, _config_zero_init, floats_tensor
+from ...test_pipeline_mixin import PipelineTesterMixin
+
+
+if is_torch_available():
+    import torch
+
+    from transformers import RTDetrForObjectDetection, RTDetrModel
+    from transformers.models.rt_detr.modeling_rt_detr import RTDETR_PRETRAINED_MODEL_ARCHIVE_LIST
+
+if is_vision_available():
+    from PIL import Image
+
+
+CHECKPOINT = "sbchoi/rtdetr_r50vd"  # TODO: replace
+
+
+class RTDetrModelTester:
+    def __init__(
+        self,
+        parent,
+        batch_size=8,
+        is_training=True,
+        use_labels=True,
+        n_targets=8,
+        num_labels=80,
+        initializer_range=0.02,
+        layer_norm_eps=1e-5,
+        batch_norm_eps=1e-5,
+        # backbone
+        use_timm_backbone=True,
+        backbone_config=None,
+        num_channels=3,
+        backbone="resnet50d",
+        use_pretrained_backbone=True,
+        dilation=False,
+        # encoder HybridEncoder
+        d_model=256,
+        encoder_in_channels=[512, 1024, 2048],
+        feat_strides=[8, 16, 32],
+        encoder_layers=1,
+        encoder_ffn_dim=64,
+        encoder_attention_heads=2,
+        dropout=0.0,
+        activation_dropout=0.0,
+        encode_proj_layers=[2],
+        pe_temperature=10000,
+        encoder_activation_function="gelu",
+        activation_function="silu",
+        eval_size=None,
+        normalize_before=False,
+        # decoder RTDetrTransformer
+        num_queries=30,
+        decoder_in_channels=[256, 256, 256],
+        decoder_ffn_dim=64,
+        num_feature_levels=3,
+        decoder_n_points=4,
+        decoder_layers=2,
+        decoder_attention_heads=2,
+        decoder_activation_function="relu",
+        attention_dropout=0.0,
+        num_denoising=10,
+        label_noise_ratio=0.5,
+        box_noise_scale=1.0,
+        learnt_init_query=False,
+        anchor_image_size=[640, 640],
+        image_size=640,
```

Comment:
This is pretty large - can we decrease this? We want out tests to be a light and fast as possible.

Reply:
For just image size, yes we can reduct the size. Does 160 seems okay to you?

Reply:
Yep! That sounds good - I think it also affects `anchor_image_size`? All the values for the tests should be as small as possible, so it's good to reduce as much as possible